Feature/add wasmjs support

amirghm opened the pull request:
✨ Add WASM & Desktop Enhancements, Bug Fixes, and Upgrades

This PR includes a variety of new features, improvements, and fixes:
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/ba238983-4f50-4dcb-955f-b42d8593cb9c" />

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/b8c10d9b-a3cd-4c19-b27b-4655d7d5067d" />

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/965cf58f-945b-4f32-b41b-90710a9c3131" />


⸻
**🧩 WASM JS Support via iframe:**
	• Load HTML from a file or raw HTML data
	• Limited support for raw links (due to iframe security restrictions)
	• JSBridge integration for cross-platform scripting
	• Basic navigation support (limited for raw links)
⸻
**💻 Desktop Improvements:**
	• Fixed file-loading issues by using a temporary folder with automatic cleanup
	• Enhanced loading UI for a smoother user experience
⸻
**🧪 HTML Sample Enhancements:**
	• Updated to demonstrate style support
	• File picker now supports media preview across all types
⸻
**🔌 Platform Integration:**
	•  Added JSBridge support to both Desktop and WASM platforms
        •  Added WebViewFileReadType to support files from both `resources/assets` and `composeResources/files` for all platforms (WasmJs only supports local `resources/assets`, and the resources need to be copied in `wasmJsMain/resources`)
	•  Added run configurations for all platforms
⸻
**🧹 Maintenance & Upgrades:**
	⬆️ Upgraded:
	•	Android Gradle Plugin (AGP)
	•	Kotlin Multiplatform (KMP)
	•	Jetpack Compose
	•	Kotlin version
	•	Other dependencies
	🗑️ Removed deprecated code
	📱 Increased Android compiled SDK to **36** and implemented EdgeToEdge screen handling
	🔧 Updated all references from KMM to KMP in strings

amirghm commented:
@KevinnZou Hey mate, I'm glad to be part of your tribe here :)
I added limited wasmJs support for your library, plus fixing some issues for the desktop, and added loading file from both **composeResources/files** and **resources/assets** for all platforms, and did some overall improvements.

Feel free to review and leave any comments, I'm so happy to resolve any upcoming problems.
Also, I'm up for fixing any upcoming issues for this library.

Best
Amir

KevinnZou reviewed (commented):
Thanks for your contribution! I have a question on this commit. Honestly, I don't quite remember why I made a specific content loading logic for Desktop, but it must be due to some loading issue on the Desktop side. Thus, I am afraid that simply removing the check will result in some bugs that will only be triggered in special cases.

amirghm commented:
> Thanks for your contribution! I have a question on this commit. Honestly, I don't quite remember why I made a specific content loading logic for Desktop, but it must be due to some loading issue on the Desktop side. Thus, I am afraid that simply removing the check will result in some bugs that will only be triggered in special cases.

Hey @KevinnZou, you are very welcome :)

We had issues with loading content on the desktop. First, I fixed the bug and also fixed loading content from files (It returns an error on loading from files), then at the end, I moved the delay right before loading the content on the desktop. I added a 500ms delay to make sure the browser is loaded successfully. If I do it without delays, it will not show anything.

So that was the reason. After fixing the issue we don't need to create different mechanism for loading the data on browser

amirghm commented:
> @amirghm There is also a code style issue, please have a look at it. Thank you so much!

It seems to be fixed now

amirghm commented:
Hey @KevinnZou Glad you liked it :)
I have some minor fixes, such as adjusting the zoom level to be consistent across all platforms.  But I can open a separate PR for them as they are not directly related to this.

For supporting files on the desktop, I first extracted the content folder and then read it from the folder. Otherwise, it wasn't possible to read files with their dependencies from the compressed file. Additionally, the folder will be cleared immediately after the browser closes.

For the wasmJs, unfortunately, the limitation is for the <iframe>, and I can't do anything with that. But I'm thinking that maybe I could use a new approach to not use iframe and instead, load the whole HTML and inject wasmJs into that one as an Iframe(reverse approach).
Anything rather than raw links, loading from data and files for wasmJs works very well, and it can help many developers (including me) to use Webview on all platforms.

I'm currently working on a cross-platform product that uses webview, so I'm actively using it, and I will fix issues once I find or add features if needed, and let you know

amirghm commented:
And feel free to update the ReadMe files for the `2.0.0` version after merging this

KevinnZou commented:
Great Jobs! I will merge it and release it soon! Thanks for your contribution!
